### ISSUE TITLE
Class Target_java_lang_invoke_TypeDescriptor_OfField[] is instantiated reflectively but was never registered

### ISSUE THREAD

**Issue by sdeleuze** [oracle/graal#3392]
With GraalVM 21.1.0, [our `discoveryclient` sample](https://github.com/spring-projects-experimental/spring-native/tree/main/samples/cloud-discovery/discoveryclient) throws this error:

```
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'loadBalancerClientFactory' defined in class path resource [org/springframework/cloud/loadbalancer/config/LoadBalancerAutoConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springframework.cloud.loadbalancer.support.LoadBalancerClientFactory]: Factory method 'loadBalancerClientFactory' threw exception; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'default.org.springframework.cloud.loadbalancer.config.LoadBalancerAutoConfiguration.LoadBalancerClientSpecification': Unsatisfied dependency expressed through constructor parameter 1: Could not convert argument value of type [[Ljava.lang.String;] to required type [[Ljava.lang.Class;]: Failed to convert value of type 'java.lang.String[]' to required type 'java.lang.Class[]'; nested exception is java.lang.IllegalArgumentException: Class com.oracle.svm.core.hub.Target_java_lang_invoke_TypeDescriptor_OfField[] is instantiated reflectively but was never registered. Register the class by using org.graalvm.nativeimage.hosted.RuntimeReflection
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:658) ~[na:na]
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:486) ~[na:na]
```

I have to add following reflection configuration to make it work:

```
[
  {
    "name": "com.oracle.svm.core.hub.Target_java_lang_invoke_TypeDescriptor_OfField[]"
  },
  {
    "name": "com.oracle.svm.core.hub.Target_java_lang_constant_Constable[]"
  },
  {
    "name": "com.oracle.svm.core.hub.Target_java_lang_invoke_TypeDescriptor[]"
  }
]
```

This is a regression and should not be needed. Notice this is a sample where we previously has [some related issue with JDK specific classes](https://github.com/spring-projects-experimental/spring-native/issues/631) with GraalVM 21.0.0. Please tag this issue with the `spring` label.

**Comment by mcraj017** [comment]
@sdeleuze thanks for reporting the issue
We will check it out and get back to you

**Comment by christianwimmer** [comment]
Fixed in f422d956ad0
